modules/pam_faillock/main.c

```diff
@@ -123,8 +126,73 @@ args_parse(int argc, char **argv, struct options *opts)
 static void
 usage(const char *progname)
 {
-	fprintf(stderr, _("Usage: %s [--dir /path/to/tally-directory] [--user username] [--reset]\n"),
-		progname);
+	fprintf(stderr,
+		_("Usage: %s [--dir /path/to/tally-directory] "
+		  " [--user username] [--reset] [--legacy-output]\n"), progname);
+}
+
+static void
+print_in_new_format(struct options *opts, struct tally_data tallies, const char *user)
+{
+	uint32_t i;
+	time_t when;
+	struct tm *tm;
+	uint16_t status;
+	char timebuf[80] = {};
```

Comment:
Why the buffer is initialized here?  It was not the case in the old code.

If you are refactoring this, consider moving into the loop all variables that are initialized and used only inside that loop.

Reply:
Will do, thanks. better to initialize variables before passing them to other functions.

Reply:
Buffers that are not expected to be accessed before being written are not required to be initialized.
When I see a buffer being initialized this way, my first thought is where the buffer is accessed before being written.